source/Outfit.cpp

```diff
@@ -712,6 +714,41 @@ const Sprite *Outfit::FlotsamSprite() const
 
 
 
+const vector<pair<const Outfit *, double>> Outfit::GetResuppliedAmmo() const
+{
+	return resuppliedAmmo;
+}
+
+
+
+bool Outfit::AmmoResupplied(const Outfit *ammo) const
+{
+	for(const pair<const Outfit *, double> &refilled : resuppliedAmmo)
+		if(ammo == refilled.first)
+			return true;
+	return false;
```

Comment:
If you use an `std::map<const Outfit *, double>`, this whole thing becomes:
`return resuppliedAmmo.contains(ammo);`